chain/bitcoind_client.go

```diff
@@ -1117,15 +1028,12 @@ func (c *BitcoindClient) filterTx(tx *wire.MsgTx,
 	// If we already know this is a relevant tx from a previous ntfn, we
 	// can shortcut the filter process and let the caller know the filter
 	// matches.
-	c.clientMtx.RLock()
-	if _, ok := c.memPool[tx.TxHash()]; ok {
-		c.clientMtx.RUnlock()
+	if _, ok := c.mempool[tx.TxHash()]; ok {
```

Comment:
godoc comment should be updated to reflect that now the caller is responsible for holding the mutex while calling this method.

Reply:
I don't think there's a need for the mutex as the map is only used within the `ntfnHandler`.

Reply:
Fixed.